Automattic/jetpack
# Fix sharing buttons block icons on simple/atomic sites

Fixes https://github.com/Automattic/jetpack/issues/34521

## Proposed changes:

This is a follow up for https://github.com/Automattic/jetpack/pull/34414. In order to address the issue, where icons are not saved and not shown on the view on Simple/Atomic sites we are moving the markup to the server side. 
As also noted in a https://github.com/Automattic/jetpack/issues/34521#issuecomment-1846453497 we are copying SVGs from https://github.com/Automattic/social-logos.
This PR also does some refactoring and removing unused code.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Use `rsync` or the cli command in order to copy changes from this PR to your sandbox.
* Modify your hosts file to point your simple/atomic site to your sandbox
* Create a new post. As an A12C you should be able to add beta blocks. Add Sharing Buttons block and individual buttons inside
* Save the post and view the results. There should be identical behavior to self hosted site like in the original pr: https://github.com/Automattic/jetpack/pull/34181
